### PR TITLE
PieChart: Refactoring & adding unsubscribe

### PR DESCRIPTION
**Pull request by torkelo**
* Fixes problem with labels being hidden behind neighboring slices
* Adds unsubscribe 

The label positions are still pretty bad but can be tweaked later

### CODE REVIEW

File: packages/grafana-ui/src/components/PieChart/PieChart.tsx [hunk 1]
```diff
@@ -153,19 +136,51 @@ export const PieChart: FC<PieChartProps> = ({
...
+    subs.add(eventBus.subscribe(DataHoverClearEvent, resetHighlightedSlice));
+
+    return () => {
+      subs.unsubscribe();
+    };
+  }, [setHighlightedTitle, eventBus]);
+
+  return highlightedTitle;
+}
+
```
Comment (hugohaggmark) [inline packages/grafana-ui/src/components/PieChart/PieChart.tsx:171]: Great refactoring, I don't think it should matter but maybe the code will be clearer if we move the arrow functions outside of the `useEffect`?
```suggestion
function useSliceHighlightState() {
  const [highlightedTitle, setHighlightedTitle] = useState<string>();
  const { eventBus } = usePanelContext();
  const setHighlightedSlice = useCallback(
    (event: DataHoverEvent) => {
      if (eventBus.isOwnEvent(event)) {
        setHighlightedTitle(event.payload.dataId);
      }
    },
    [eventBus]
  );
  const resetHighlightedSlice = useCallback(
    (event: DataHoverClearEvent) => {
      if (eventBus.isOwnEvent(event)) {
        setHighlightedTitle(undefined);
      }
    },
    [eventBus]
  );

  useEffect(() => {
    if (!eventBus) {
      return;
    }

    const subs = new Subscription();
    subs.add(eventBus.subscribe(DataHoverEvent, setHighlightedSlice));
    subs.add(eventBus.subscribe(DataHoverClearEvent, resetHighlightedSlice));

    return () => {
      subs.unsubscribe();
    };
  }, [setHighlightedSlice, resetHighlightedSlice, eventBus]);

  return highlightedTitle;
}
```

File: packages/grafana-ui/src/components/PieChart/PieChart.tsx [hunk 2]
```diff
@@ -130,7 +113,7 @@ export const PieChart: FC<PieChartProps> = ({
     fieldConfig,
     reduceOptions,
     data,
-    theme,
+    theme: theme.v1,
```
Comment (hugohaggmark) [inline packages/grafana-ui/src/components/PieChart/PieChart.tsx:116]: nit: Maybe the `getLegend` arrow function here should also be a hook?

**Review by hugohaggmark — APPROVED** [review]
Code looks great, I haven't had time to test it.

File: packages/grafana-ui/src/components/PieChart/PieChart.tsx (hunk 1 above)
Comment (dprokop) [reply packages/grafana-ui/src/components/PieChart/PieChart.tsx:171]: I agree with Hugo, he's proposal looks cleaner

**Review by dprokop — APPROVED** [review]
LGTM

File: packages/grafana-ui/src/components/PieChart/PieChart.tsx (hunk 1 above)
Comment (torkelo) [reply packages/grafana-ui/src/components/PieChart/PieChart.tsx:171]: I tested this but means two additional dependencies to the useEffect, which makes the code more complex and the function scoping more complex

File: packages/grafana-ui/src/components/PieChart/PieChart.tsx (hunk 2 above)
Comment (torkelo) [reply packages/grafana-ui/src/components/PieChart/PieChart.tsx:68]: moved this out to a separate function

**Pull request merged**